File: practical-bdd/java/pages/BasePage.java
package pages;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.support.PageFactory;

abstract class BasePage {

    WebDriver driver;

    BasePage(WebDriver driver) {
        this.driver = driver;
        PageFactory.initElements(this.driver, this);
    }

}

File: practical-bdd/c-sharp/PageObjects/BasePage.cs
using OpenQA.Selenium;
using OpenQA.Selenium.Support.UI;

namespace Polteq.Training.Practical.BDD.PageObjects
{
    public abstract class BasePage
    {
        protected IWebDriver driver;
        protected WebDriverWait wait;

        public BasePage(IWebDriver driver)
        {
            this.driver = driver;
            wait = new WebDriverWait(this.driver, TimeSpan.FromSeconds(10));
        }
    }
}
